apps/site/util/downloadUtils/index.tsx

```diff
@@ -0,0 +1,215 @@
+  compatibility: Partial<DownloadCompatibility>;
+} & Omit<SelectValue<T>, 'label'>;
+
+// This function is used to get the next item in the dropdown
+// when the current item is disabeld/excluded/not valid
+// And this is useful when a parent release value (i.e. OS) is changed
+// and requires the current dropdown (i.e. Platform) to be updated
+export const nextItem = <T extends string>(
+  current: T,
+  items: Array<DownloadDropdownItem<T>>
+): T => {
+  const item = items.find(({ value }) => String(value) === String(current));
+
+  const isDisabledOrExcluded = !item || item.disabled;
+
+  if (isDisabledOrExcluded) {
+    const nextItem = items.find(({ disabled }) => !disabled);
+
+    if (nextItem) {
+      return nextItem.value;
+    }
+  }
+
+  return current;
+};
+
+// This function is used to parse the compatibility of the dropdown items
+// In a nice and static way that allows a lot of abstraction and flexibility
+export const parseCompat = <
+  K extends string,
+  T extends DownloadDropdownItem<K>,
+>(
+  items: Array<T>,
+  { os, installMethod, platform, version, release }: Types.ReleaseContextType
+): Array<T> => {
+  const satisfiesSemver = (semver: string) => satisfies(version, semver);
+
+  const supportsOS = (i: T['compatibility']) => i.os?.includes(os) ?? true;
+
+  const supportsInstallMethod = (i: T['compatibility']) =>
+    i.installMethod?.includes(installMethod) ?? true;
+
+  const supportsPlatform = (i: T['compatibility']) =>
+    i.platform?.includes(platform) ?? true;
+
+  const supportsVersion = (i: T['compatibility']) =>
+    i.semver?.some(satisfiesSemver) ?? true;
+
+  const supportsRelease = (i: T['compatibility']) =>
+    i.releases?.includes(release.status) ?? true;
+
+  return items.map(item => ({
+    ...item,
+    disabled:
+      !supportsOS(item.compatibility) ||
+      !supportsInstallMethod(item.compatibility) ||
+      !supportsPlatform(item.compatibility) ||
+      !supportsVersion(item.compatibility) ||
+      !supportsRelease(item.compatibility),
+  }));
+};
+
+// Here the list of Operating System Dropdown items are defined !
+export const OPERATING_SYSTEMS: Array<DownloadDropdownItem<UserOS>> = [
+  {
+    label: osConstants[0].label,
+    value: osConstants[0].value as UserOS,
+    compatibility: osConstants[0].compatibility,
+    iconImage: <OSIcons.Microsoft width={16} height={16} />,
+  },
+  {
+    label: osConstants[1].label,
+    value: osConstants[1].value as UserOS,
+    compatibility: osConstants[1].compatibility,
+    iconImage: <OSIcons.Apple width={16} height={16} />,
+  },
+  {
+    label: osConstants[2].label,
+    value: osConstants[2].value as UserOS,
+    compatibility: osConstants[2].compatibility,
+    iconImage: <OSIcons.Linux width={16} height={16} />,
+  },
+  {
+    label: osConstants[3].label,
+    value: osConstants[3].value as UserOS,
+    compatibility: osConstants[3].compatibility,
+    iconImage: <OSIcons.AIX width={16} height={16} />,
+  },
+];
```

Comment:
You can store the icon in the JSON.

i.e. in the JSON do `icon: AIX`, and then in the JS, map it to `OSIcons[icon]`